src/libraries/System.Text.RegularExpressions/src/System/Text/SegmentStringBuilder.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    /// <summary>Provides a value type string builder composed of individual segments represented as <see cref="ReadOnlyMemory{T}"/> instances.</summary>
+    [DebuggerDisplay("Count = {_count}")]
+    internal struct SegmentStringBuilder
```

Comment:
Probably worth adding a few tests?

Reply:
Doesn't seem valuable enough to me; we don't have tests for most other internal implementation details.  It's just a helper for the Replace implementation; if it's buggy in a way that impacts Replace, that'll show up in tests for Replace.